app/models/candidate_interface/degrees_form.rb

```diff
@@ -15,17 +15,51 @@ class DegreesForm
 
     validate :award_year_is_date, if: :award_year
 
-    def self.build_from_application(application_form)
-      application_form.application_qualifications.degrees.map do |degree|
+    class << self
+      def build_from_application(application_form)
+        application_form.application_qualifications.degrees.map do |degree|
+          new(
+            id: degree.id,
+            qualification_type: degree.qualification_type,
+            subject: degree.subject,
+            institution_name: degree.institution_name,
+            grade: degree.grade,
+            predicted_grade: degree.predicted_grade,
+            award_year: degree.award_year,
+          )
+        end
+      end
+
+      def find_using_application(application_form, degree_id)
```

Comment:
what the method does is not in line with the name `find_using_application` (it is just a small part),
in case we are already doing this way, do not consider this comment, but we should plan to refactor this.